src/Propulsion/Internal.fs

```diff
@@ -63,10 +64,10 @@ module Channel =
     let unboundedSw<'t> = Channel.CreateUnbounded<'t>(UnboundedChannelOptions(SingleWriter = true))
     let unboundedSwSr<'t> = Channel.CreateUnbounded<'t>(UnboundedChannelOptions(SingleWriter = true, SingleReader = true))
     let boundedSw<'t> c = Channel.CreateBounded<'t>(BoundedChannelOptions(c, SingleWriter = true))
-    let waitToWrite (w : ChannelWriter<_>) ct = w.WaitToWriteAsync(ct)
+    let waitToWrite (w : ChannelWriter<_>) ct = w.WaitToWriteAsync(ct).AsTask() :> Task
     let tryWrite (w : ChannelWriter<_>) = w.TryWrite
     let write (w : ChannelWriter<_>) = w.TryWrite >> ignore
-    let inline awaitRead (r : ChannelReader<_>) ct = let vt = r.WaitToReadAsync(ct) in vt.AsTask()
+    let inline awaitRead (r : ChannelReader<_>) ct = r.WaitToReadAsync(ct).AsTask()
```

Comment:
does this not trip a `/w5` ?

Reply:
I don't know what a `/w5` is so can't say 😓

Reply:
Thought there was a WarningLevel5 in the src Directory.build.props and that it complains about a defensive copy of. a struct

Reply:
Oh, if it's not a red ass error in my face I doubt I'd ever notice. I'll run the build and see

Reply:
<img width="260" alt="Screenshot 2023-06-11 at 15 32 57" src="https://github.com/jet/propulsion/assets/3721521/37ec6b44-e829-4851-812f-bc75615aed63">
Looks fine

Reply:
hm, maybe there was one once but def not now, and it does seem like w5 is on. OK, all good!

Reply:
should prob turn on `warningsaserrors`

Reply:
wait, a rebuild does trigger it  -best to turn on warningsaserrors to stop a regression

Reply:
Can you add this please ?